gh-5: Adds basic tests for app and browserWindow components.

colinbdclark opened the pull request.

colinbdclark committed:
gh-5: Promisifies the Electron app's ready event.

colinbdclark committed:
gh-5: Adds basic tests for app and browserWindow components.

colinbdclark committed:
gh-5: Bumps package version in order to publish a dev release.

colinbdclark committed:
gh-5: Cleanup and linting.

amb26 commented on src/main-process/window.js:
```diff
@@ -24,23 +23,44 @@ fluid.defaults("electron.browserWindow", {
+    },
+
+    invokers: {
+        show: {
```
This can probably be axed, esp. since "hide" was never implemented. If we have modelised some part of application function, this serves as a preferable interface to "invokers" which will be abolished in any case

amb26 commented:
Test run on Windows hangs at:
````
17:26:54.042:  jq: Test concluded - Module "App creation tests" Test name "App onReady": 2/2 passed - PASS
17:26:59.045:  Test case listener has not responded after 5000ms - at sequence pos 1 of 1 sequence element {
    "event": "{app windough}.events.afterShow",
    "listener": "electron.tests.windowTester.bounds",
    "args": [
        "{app}.windough"
    ]
} of fixture Bounds
````

amb26 commented on tests/unit/js/window-tests.js:
```diff
@@ -0,0 +1,138 @@
+                    name: "Bounds",
+                    sequence: [
+                        {
+                            event: "{app windough}.events.afterShow",
```
Standard problem here - the afterShow event may fire synchronously before the sequence begins to bind to it. After we've merged the "new framework" impl we can improve the IoC Testing Framework so that it can win these construction races but in the meantime you will have to add some annoying boilerplate to every sequence to explicitly construct, e.g. the "windough" triggering an explicit "createOnEvent" for it.

colinbdclark committed:
Only hides a window if it was already showing.
Tests now appear to stall consistently across platforms, still need to be fixed.

colinbdclark committed:
Moves test app bounds from model to options.

colinbdclark committed:
gh-5: Fixes race condition with window tests.
Adds test for isShowing and fixes bugs.

colinbdclark committed:
gh-5: Fixes broken tests on Windows.

colinbdclark commented:
Thanks very much for your review, @amb26. I've addressed your comments, and fixed the race condition that was causing the Windows tests to stall, and then fixed all issues related to test failures on Windows.

Do you think this PR is ready for a merge?

colinbdclark committed:
Moves Electron to devDependencies and updates to latest version.
